Makefile

```diff
@@ -38,6 +44,7 @@ ensure-submodule-initialized:
 	else \
 		echo "Submodule found successfully"; \
 	fi
+	cd viam-cartographer/cartographer && git checkout . && git apply ../cartographer_patches/carto.patch
```

Comment:
Nice! I like the placement of this patch

Reply:
What if the ensure submodule command is run twice? Will the second patch cause issues?

Reply:
No, that's why it does a git checkout first... that reverts to the unpatched state. Also, if you make further modifications to ANYTHING in carto, you can quickly save them with just `git diff --output ../cartographer_patches/carto.diff` from within the cartographer submodule directory.